### PR TITLE
feat(DTFS2-7616): remove dev and feature from QAT

### PR DESCRIPTION
**Pull request by abhi-markan**
## Introduction :pencil2:
Ensure effecient deployment on `dev` and `feature`.

## Resolution :heavy_check_mark:
* Remove `dev` and `feature` from pre-requiste QAT since QAT will still be executed on `staging` and `prod` enviornments.
* Automatic `main` to `dev` depoyment at 00:00 every day.
* Introduced GitHub composite modules for re-usability and abstractable codebase.

## Miscellaneous :heavy_plus_sign:
* Dependencies updates.
* Updated express from `4.21.1` to `4.21.2` and path-to-regexp from `0.1.10` to `0.1.12` to address [CVE-2024-52798](https://www.mend.io/vulnerability-database/CVE-2024-52798)

### CODE REVIEW

File: .github/actions/webapp/action.yml [hunk 1]
```diff
@@ -0,0 +1,116 @@
+name: 'Deploy Azure Functions'
```
Comment (Zainzzkk) [inline .github/actions/webapp/action.yml:1]: documentation at the top like the other files?

File: .github/actions/webapp/action.yml [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+description: 'This custom GitHub Actions module deploys from defined source to defined Azure Functions microservice'
```
Comment (ChristianMcc) [inline .github/actions/webapp/action.yml:2]: This file is called webapp/action.yml but looks like it's duplicating the azure functions deployment

**Review by ChristianMcc — CHANGES_REQUESTED** [review]

File: .github/actions/mssql/action.yml [hunk 3]
```diff
@@ -0,0 +1,52 @@
+# This GitHub Action is designed to deploy the latest data migration to an MSSQL database on an AMI.
```
Comment (ChristianMcc) [inline .github/actions/mssql/action.yml:1]: Can we change this to 'latest schema migration'

File: .github/actions/mssql/action.yml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+name: 'Deploy MSSQL'
```
Comment (ChristianMcc) [inline .github/actions/mssql/action.yml:3]: Is it worth calling this MSSQL Migrations? This makes it sound like we are deploying a database not just running schema based migrations

File: .github/actions/mssql/action.yml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+description: 'Deploy latest data migration to MSSQL on AMI'
```
Comment (ChristianMcc) [inline .github/actions/mssql/action.yml:4]: ditto on removing the word data, can include 'schema' if we want to be explicit

File: .github/actions/mssql/action.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  username:
+    description: 'MSSQL connection username'
+    required: true
+  password:
+    description: 'MSSQL connection password'
+    required: true
+  database:
+    description: 'MSSQL database name'
+    required: true
+  log:
```
Comment (ChristianMcc) [inline .github/actions/mssql/action.yml:23]: I don't think we need this log environment variable for migrations

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -0,0 +1,154 @@
+# Release Workflow
+# ****************
+#
+# This GitHub Actions workflow automates the release process for deploying web apps and Azure Functions to different environments.
+# It is triggered by the completion of the QAT workflow and runs when there is a push to a release branch (matching the pattern `release-*` but excluding `release-please*`).
+# The workflow handles the deployment of SQL migrations, web apps, and Azure functions sequentially.
+#
+# Key Stages:
+# 1. **Setup**: Determines the appropriate environment based on the release branch and configures environment variables (e.g., timezone).
```
Comment (ChristianMcc) [inline .github/workflows/release.yml:9]: extra comma at e.g.

File: .github/workflows/release.yml [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+# 2. **SQL Database Migration**: Deploys SQL migrations to the specified environment.
```
Comment (ChristianMcc) [inline .github/workflows/release.yml:10]: Runs SQL migrations*

File: .github/workflows/release.yml [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+# 3. **WebApp Deployment**: Sequentially deploys a list of web apps defined in the matrix (`portal-ui`, `gef-ui`, etc.) to the target environment.
+# 4. **Function Deployment**: Sequentially deploys a list of Azure functions (e.g., `acbs-function`) to the target environment.
+#
+# Key Points:
+# - The workflow is triggered by the completion of the QAT workflow.
+# - The deployment environment is determined dynamically based on the release branch (e.g., staging or prod).
```
Comment (ChristianMcc) [inline .github/workflows/release.yml:16]: extra , after e.g.

**Review by ChristianMcc — COMMENTED** [review]
left a few more comments/nit-picks mostly about migrations

File: .github/actions/functions/action.yml [hunk 10]
```diff
@@ -0,0 +1,118 @@
+# This GitHub Action is designed to deploy the code bases from specified branch to an Azure Functions
+
+name: 'Deploy Azure Functions'
+description: 'This custom GitHub Actions module deploys from defined source to defined Azure Functions microservice'
+
+# Define the inputs required for this action.
```
Comment (Zainzzkk) [inline .github/actions/functions/action.yml:6]: Not a blocker, this should be defines? (can be done later)

File: gef-ui/package.json [hunk 11]
```diff
@@ -51,7 +51,7 @@
     "date-fns": "^2.30.0",
     "date-fns-tz": "^2.0.1",
     "dotenv": "16.0.3",
-    "express": "^4.21.1",
+    "express": "^4.21.2",
```
Comment (Zainzzkk) [inline gef-ui/package.json:54]: if we can't upgrade this to the latest version (if breaking), we could use `~`?

**Review by Zainzzkk — APPROVED** [review]

File: gef-ui/package.json (hunk 11 above)
Comment (abhi-markan) [reply gef-ui/package.json:54]: They are not breaking changes, axios is.

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**